pkg/sentry/fsimpl/fuse/connection.go

```diff
@@ -118,13 +261,51 @@ func (conn *Connection) NewRequest(creds *auth.Credentials, pid uint32, ino uint
 }
 
 // Call makes a request to the server and blocks the invoking task until a
-// server responds with a response.
-// NOTE: If no task is provided then the Call will simply enqueue the request
-// and return a nil response. No blocking will happen in this case. Instead,
-// this is used to signify that the processing of this request will happen by
-// the kernel.Task that writes the response. See FUSE_INIT for such an
-// invocation.
-func (conn *Connection) Call(t *kernel.Task, r *Request) (*Response, error) {
+// server responds with a response. Task should never be nil.
+// For async tasks, use CallAsync().
+func (conn *connection) Call(t *kernel.Task, r *Request) (*Response, error) {
+	// Block requests sent before connection is initalized.
+	if !conn.Initialized() {
+		if err := t.Block(conn.initializedChan); err != nil {
+			return nil, err
+		}
+	}
+
+	return conn.call(t, r)
+}
+
+// CallAsync makes an async (aka background) request.
+// Those requests either do not expect a response (e.g. release) or
+// the response should be handled by others (e.g. init).
+// Return immediately unless the connection is blocked.
+// Async call example: init, release, forget, aio, interrupt.
+//
+// t can be nil if and only if the request should not be blocked
```

Comment:
I think somethings wrong here. The intention with this is that task can never be nil here either. The idea is that we will block using `t` only if it hasn't been initialized. This blocking doesn't happen anyway if the operation is a FUSE_INIT one. We should not allow t to be nil here at all.

`call` is the one that can be called with a nil task and this should be the only function that invokes `call` with a nil task.

Reply:
Right it is a better solution